src/uct/cuda/cuda_copy/cuda_copy_iface.h

```diff
@@ -48,23 +49,30 @@ typedef struct uct_cuda_copy_queue_desc {
     ucs_queue_elem_t            queue;
 } uct_cuda_copy_queue_desc_t;
 
+typedef struct uct_cuda_copy_per_ctx_rsc {
+    CUcontext                   cuda_ctx;
+    unsigned long long          ctx_id;
+    /* pool of cuda events to check completion of memcpy operations */
+    ucs_mpool_t                 cuda_event_desc;
```

Comment:
do we really need to have this mpool per context? Maybe one common mpool is enough?

Reply:
No event and stream resources are associated with a context.